modules/tinymce/src/themes/silver/test/ts/headless/components/imagepreview/BasicImagePreviewTest.ts

```diff
@@ -0,0 +1,118 @@
+import { ApproxStructure, Assertions, UiFinder } from '@ephox/agar';
+import { AlloyComponent, GuiFactory, Representing, TestHelpers } from '@ephox/alloy';
+import { describe, it } from '@ephox/bedrock-client';
+import { Optional } from '@ephox/katamari';
+import { Ready } from '@ephox/sugar';
+import { assert } from 'chai';
+
+import { ImagePreviewDataSpec, renderImagePreview } from 'tinymce/themes/silver/ui/dialog/ImagePreview';
+
+describe('headless.tinymce.themes.silver.components.imagepreview.BasicImagePreviewTest', () => {
+  const testImageUrl = 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8z8DwHwAFBQIAX8jx0gAAAABJRU5ErkJggg==';
+
+  const hook = TestHelpers.GuiSetup.bddSetup((_store, _doc, _body) => GuiFactory.build(
+    {
+      dom: {
+        tag: 'div',
+        styles: {
+          width: '200px'
+        }
+      },
+      components: [
+        renderImagePreview({
+          name: 'test-panel',
+          height: Optional.some('200px'),
+        }, Optional.none())
+      ]
+    }
+  ));
+
+  const findImage = (component) => UiFinder.findIn<HTMLImageElement>(component.element, 'img').getOrDie();
+
+  const assertImageState = (label: string, component: AlloyComponent, url: string) => {
+    const node = findImage(component).dom;
+    assert.equal(node.src, url, label + ' - checking "src" attribute');
+  };
+
+  const setPanelState = (component: AlloyComponent, state: ImagePreviewDataSpec) => {
+    // the component is a wrapper div just there to set a known width, the image preview is the child
+    const [ imagePreview ] = component.components();
```

Comment:
This really should be using a memento to get access to the image preview element instead of relying on it being a child and being the first item.

Reply:
I tried that, but the outer component can't be a memento. I think I mentioned this in Slack - the UID seems to be overwritten during the render process. I think the only way to make it a memento is to add an empty div shell to the component, and have the memento inside.

Reply:
You don't make the outer component a memento you make the image preview component a memento. I'll give it a shot locally and make sure it works.

Reply:
the image preview component _is_ the outer component?

Reply:
Oh sorry I forgot this was in the test. Yes, the outer component is there to cap the width of the image preview (since we removed the width config).

I wonder if we could add `Representing` to the outer component and have it redirect set value calls to the preview.

Reply:
Already mentioned this via DMs, but dumping this here as well. This is what I was referring to and seems to work:
```
  const memImagePreview = Memento.record(renderImagePreview({
    name: 'test-panel',
    height: Optional.some('200px'),
  }, Optional.none()));

  const hook = TestHelpers.GuiSetup.bddSetup((_store, _doc, _body) => GuiFactory.build(
    {
      dom: {
        tag: 'div',
        styles: {
          width: '200px'
        }
      },
      components: [
        memImagePreview.asSpec()
      ]
    }
  ));

  const findImage = (component: AlloyComponent) => UiFinder.findIn<HTMLImageElement>(component.element, 'img').getOrDie();

  const assertImageState = (label: string, component: AlloyComponent, url: string) => {
    const node = findImage(component).dom;
    assert.equal(node.src, url, label + ' - checking "src" attribute');
  };

  const setPanelState = (component: AlloyComponent, state: ImagePreviewDataSpec) => {
    const imagePreview = memImagePreview.get(component);
    Representing.setValue(imagePreview, state);
  };
```

Reply:
Yes, it works, I just didn't get a chance to commit and push before lunch :)